client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/BasicCache.java

```diff
@@ -49,11 +49,21 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(BasicCache.class);
 
+    /**
+     * Maximum age for a cached connection check result to be used in {@link #isConnected()}.
+     * Value is chosen to be a bit longer than the default Kubernetes readiness probe interval, so that
+     * the periodic invocations of {@link #checkForCacheAvailability()} as part of the readiness check
+     * keep the cached connection check result up to date.
+     */
+    private static final long CACHED_CONNECTION_CHECK_RESULT_MAX_AGE_MILLIS = TimeUnit.SECONDS.toMillis(11);
```

Comment:
I would suggest to use Duration, rather than a plain long.

Reply:
This effectively sets a lower boundary to the k8s readiness/liveness check frequency, doesn't it? I wonder if we could simply continuously update the connection status with each invocation of any of the cache's methods, i.e. whenever a method successfully interacts with the cache, we set the flag to true and set it to false if the invocation fails. I do not think that this would be any worse than doing the periodic put triggered by the readiness probe which doesn't guarantee that the connection works for subsequent method invocations anyway. On startup of the cache we invoke the put method once to initialize the connection status flag. WDYT?

Reply:
Yes, I think that would be a better solution still and would also simplify things in the `BasicCache` class.
Checking again, if with the newer Hotrod client there is maybe a way to use `remoteCache.getRemoteCacheManager().getConnectionCount()` or something like that, in addition to the cache operation status update, would also be good (haven't tested that yet).

I think that would better be done in a new PR, closing this here then.